### PR TITLE
Refactor keadm E2E shell script #5266

### PR DESCRIPTION
**Pull request by NishantBansal2003**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Optimize the script by removing redundant functions such as `prepare_cluster` and `run_test`, and extract commonly used variables in the `keadm_common_e2e.sh` file for easier maintenance.

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubeedge/kubeedge/issues/5266

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: tests/scripts/keadm_common_e2e.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The KubeEdge Authors.
```
Comment (Shelley-BaoYue) [inline tests/scripts/keadm_common_e2e.sh:3]:
```suggestion
# Copyright 2024 The KubeEdge Authors.
```

**Review by Shelley-BaoYue — COMMENTED** [review]
Great work! Welcome to KubeEdge community!

**Comment by Shelley-BaoYue** [comment]
Some errors occur in keadm e2e test, please help solve it. : )

**Comment by NishantBansal2003** [comment]
Sure, Can you point me to some resource where I can learn and fix the particular tests?

**Comment by Shelley-BaoYue** [comment]
> Sure, Can you point me to some resource where I can learn and fix the particular failing tests?

Error information:  https://github.com/kubeedge/kubeedge/actions/runs/9825030740/job/27194820195?pr=5709
You can debug it in the local environment or add debugging steps in the CI according to the https://github.com/mxschmitt/action-tmate .

**Comment by NishantBansal2003** [comment]
Hey @Shelley-BaoYue, I tried hard to debug the failing check but was not able to reproduce and find the root cause of the error. Can you please help me a little? Also, some dependencies are unavailable for macOS M1, like CRI or containerd. Maybe due to these, I am not able to identify the error.

**Comment by NishantBansal2003** [comment]
Hey @Shelley-BaoYue, Can you review the changes? Also, please trigger the Github checks so I can fix the issues if a test fails.

**Comment by NishantBansal2003** [comment]
Hey @Shelley-BaoYue, Can you please trigger the Github checks so I can fix the issues if a test fails?

**Comment by NishantBansal2003** [comment]
Hey @Shelley-BaoYue, All the test pass, Please review the changes and let me know if there is need to add/change something.
Thanks

**Comment by NishantBansal2003** [comment]
PTAL...

**Review by Shelley-BaoYue — COMMENTED** [review]
/lgtm 
It would be better if you can squash these commits into one. 😄

**Pull request closed without merging**